near-lake-primitives/src/types/receipts.rs

```diff
@@ -0,0 +1,169 @@
+use crate::near_indexer_primitives::{
+    types::{AccountId, Balance, Gas},
+    views, CryptoHash, IndexerExecutionOutcomeWithReceipt,
+};
+use near_crypto::PublicKey;
+use near_primitives_core::serialize::{base64_format, dec_format};
+
+#[derive(Debug, Clone)]
+pub struct ExecutedReceipt {
```

Comment:
I am thinking about renaming it to just `Receipt` as I also want to provide `included_receipts` to the `Block` and executed ones with included will differ only by the status.

I am open to propositions on how to differentiate `ActionReceipts` from `DataReceipts` not following the `ReceiptEnumView` pattern.

Reply:
I would consider making `Receipt` to be an enum:

```rust
enum Receipt {
    ActionReceipt(ActionReceipt),
    DataReceipt(DataReceipt),
}
```

Alternatively, I would consider splitting them into separate vectors: `action_receipts: Vec<ActionReceipt>` and `data_receipts: Vec<DataReceipt>` - I believe this is the most straightforward way as I have not yet seen a case where users need to deal with a mix of action and data receipts.

Reply:
@frol thanks.

The idea is great. Though I would need to have something like:

- `.action_receipts_included()`
- `.data_receipts_included()`
- `.receipts_executed()` (`DataReceipt` is non-executable)

Though I already have the `.actions()` method that would return the collection of `Action` structures which is sort of `ActionReceipt` but concentrated on the action side more than on the receipt side of it. It might be confusing. I need to think a bit more about this. 

Thank you for your inputs!